picard/ui/coverartbox.py

```diff
@@ -545,6 +550,43 @@ def load_remote_image(self, url, data):
             log.warning("Can't load image: %s", e)
             return
 
+    def delete_cover_art(self):
+        if not self.item or not self.item.metadata.images:
```

Comment:
There's an important flaw here:

- if an album was just loaded, and no image was downloaded (it has only embedded artwork), actual metadata that contains images is `self.item.orig_metadata` and `self.item.metadata` is empty.